src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -2282,13 +2282,22 @@ private bool TryProcessEnvChange(int tokenLength, TdsParserStateObject stateObj,
                                 return false;
                             }
 
-                            // give the parser the new collation values in case parameters don't specify one
-                            _defaultCollation = env.newCollation;
-                            int newCodePage = GetCodePage(env.newCollation, stateObj);
-                            if (newCodePage != _defaultCodePage)
+                            // UTF8 collation
+                            if ((env.newCollation.info & TdsEnums.UTF8_IN_TDSCOLLATION) == TdsEnums.UTF8_IN_TDSCOLLATION)
                             {
-                                _defaultCodePage = newCodePage;
-                                _defaultEncoding = System.Text.Encoding.GetEncoding(_defaultCodePage);
+                                _defaultEncoding = Encoding.UTF8;
+                                _defaultCollation = env.newCollation;
```

Comment:
`_defaultCollation = env.newCollation`
This is set in both if and else. Can we move it out to before if and else.

Reply:
Moved it out before if and else as requested.